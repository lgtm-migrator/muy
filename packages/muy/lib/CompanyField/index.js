import React from "react"
import TextField from "@material-ui/core/TextField"

const CompanyField = ({
  id,
  inputProps,
  maxLength,
  name,
  required,
  ...props
}) => (
  <TextField
    aria-required={required}
    id={id}
    inputProps={{ "data-name": name, maxLength, ...inputProps }}
    name={name}
    required={required}
    type={"text"}
    {...props}
  />
)

export default CompanyField

CompanyField.defaultProps = {
  id: "company",
  inputProps: {},
  maxLength: 256,
  name: "company",
}
