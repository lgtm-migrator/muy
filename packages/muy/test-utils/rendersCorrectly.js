import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
