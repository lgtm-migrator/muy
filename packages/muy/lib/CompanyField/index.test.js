import React from "react"
import CompanyField from "."
import rendersCorrectly from "../../test-utils/rendersCorrectly"

describe("<CompanyField /> component", () => {
  it("renders correctly", () => {
    rendersCorrectly(<CompanyField />)
  })
})
