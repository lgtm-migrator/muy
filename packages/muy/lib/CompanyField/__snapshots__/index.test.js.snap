// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompanyField /> component renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      data-name="company"
      disabled={false}
      id="company"
      maxLength={256}
      name="company"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;
